### PR TITLE
Adding some error handling to getting the whitelists

### PR DESCRIPTION
**Pull request by solarce**
- Uses Faraday's error handling
- Splits the URLs into two parts
  ( will mean some config updates when pushing)

### CODE REVIEW

File: lib/travis/build/addons/apt.rb [hunk 1]
```diff
@@ -41,19 +52,25 @@ def load_source_whitelist
...
           end
 
           def fetch_source_whitelist
-            Faraday.get(source_whitelist_url).body.to_s
+            get_whitelist(source_whitelist_url)
           end
 
           def package_whitelist_url
-            ENV['TRAVIS_BUILD_APT_PACKAGE_WHITELIST'] || ENV['TRAVIS_BUILD_APT_WHITELIST']
+            url_base = ENV['TRAVIS_BUILD_APT_BASE_URL']
```
Comment (cotsog) [inline lib/travis/build/addons/apt.rb]: I'm curious, where are these env variables defined? e.g. TRAVIS_BUILD_APT_PACKAGE_WHITELIST, TRAVIS_BUILD_APT_BASE_URL, TRAVIS_BUILD_APT_PACKAGE_WHITELIST_PATH, etc.

Comment (solarce) [inline lib/travis/build/addons/apt.rb]: They are defined in the Heroku app's config right now

Comment (cotsog) [inline lib/travis/build/addons/apt.rb]: Ah! Got it! Thanks!

**Comment by BanzaiMan** [comment]
@solarce What coordination is necessary to push this into production?

**Comment by solarce** [comment]
@BanzaiMan there are a couple of things.

1.There are some config values that need to be changed on staging and prod when it's pushed.
  - add `TRAVIS_BUILD_APT_BASE_URL` and set it to `https://raw.githubusercontent.com` ( _no trailing `/`_ )
  - add `TRAVIS_BUILD_APT_SOURCE_WHITELIST_PATH` and set it to `/travis-ci/apt-source-whitelist/master/ubuntu.json`
  - add `TRAVIS_BUILD_APT_PACKAGE_WHITELIST_PATH` and set it to `/travis-ci/apt-package-whitelist/master/ubuntu-precise`
  - once these new variables are confirmed working in staging and prod, the old ones should be removed

2. I was gonna try to figure out how to test it locally and that's where I left off. If you've got time to push it to staging and test it out, that'd be excellent

3. We should discuss moving the actual `.json` we fetch to S3 and setting up a deployment phase in a build to push to S3

**Comment by meatballhat** [comment]
No longer applicable! :heart:

**Pull request closed without merging**